src/secrets/src/cache.rs

```diff
@@ -0,0 +1,378 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::collections::BTreeMap;
+use std::fmt;
+use std::sync::atomic::{AtomicBool, AtomicU64, Ordering};
+use std::sync::{Arc, RwLock};
```

Comment:
Does it matter that we're using the sync `RwLock` when this will be used from async contexts? Not a criticism of the code, I just want to know for myself.

Reply:
No, it's all good to use the `std::sync` locks as long as you don't hold a lock across an `await` point! (There's a Clippy lint for this now, which makes it a lot less stressful.) Recommend this section of the Tokio docs, which explains the situation really well! https://docs.rs/tokio/latest/tokio/sync/struct.Mutex.html#which-kind-of-mutex-should-you-use